src/components/configuration-tree/configuration-tree.component.jsx

```diff
@@ -22,16 +17,35 @@ import styles from './configuration-tree.styles.js'
 
 const useStyles = createUseStyles(styles, { name: 'ConfigurationTree' })
 
-const ConfigurationTree = ({ siteId, id, name, value, error, branches, tooltip, setConfigurationStatus, setDataflowVariableValue, setDataflowVariableValues, t }) => {
+const ConfigurationTree = ({
+  siteId,
+  id,
+  name,
+  value,
+  error,
+  branches,
+  tooltip,
+  setConfigurationStatus,
+  setDataflowVariableValue,
+  setDataflowVariableValues,
+  setRbaRulesMergeOrReplace,
+  t,
+}) => {
   const dispatch = useDispatch()
   const classes = useStyles()
 
   const [isMouseOverIcon, setIsMouseOverIcon] = useState(false)
   const [tooltipTarget, setTooltipTarget] = useState('')
+  const [isRBAChecked, setIsRBAChecked] = useState(false)
 
   const onCheckBoxStateChangeHandler = (event) => {
     const checkBoxId = event.srcElement.id
     const value = event.srcElement.checked
+
+    if (checkBoxId === 'rba' || checkBoxId === 'RBA Rules') {
```

Comment:
I don't like to see these specific if statements that increases the complexity and the maintenance costs of the code.
There is no other way, like creating a different component for this kind of configuration to encapsulate the behavior? See the dataflow-settings component to inspire...